lib/util/node-event-generator.js

```diff
@@ -37,6 +130,16 @@ class NodeEventGenerator {
      */
     enterNode(node) {
         this.emitter.emit(node.type, node);
+
+        if (node.parent) {
+            this.currentAncestry.unshift(node.parent);
```

Comment:
I just tried it. It looks like the performance is roughly the same either way, so I'll leave `.shift` and `.unshift` for better readability.